authorization-common/src/main/java/com/sequenceiq/authorization/utils/EventAuthorizationUtils.java

```diff
@@ -0,0 +1,90 @@
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.authorization.resource.AlternativeAuthorizationResourceActionProvider;
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.authorization.service.CommonPermissionCheckingUtils;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+
+@Component
+public class EventAuthorizationUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventAuthorizationUtils.class);
+
+    private AlternativeAuthorizationResourceActionProvider actionAlternativeProvider;
+
+    private CommonPermissionCheckingUtils permissionCheckingUtils;
+
+    public EventAuthorizationUtils(CommonPermissionCheckingUtils permissionCheckingUtils,
+            AlternativeAuthorizationResourceActionProvider actionAlternativeProvider) {
+        this.actionAlternativeProvider = actionAlternativeProvider;
+        this.permissionCheckingUtils = permissionCheckingUtils;
+    }
+
+    public void checkPermissionBasedOnResourceTypeAndCrn(Collection<EventAuthorizationDto> eventAuthorizationDtos) {
+        throwIfNull(eventAuthorizationDtos,
+                () -> new IllegalArgumentException("The collection of " + EventAuthorizationDto.class.getSimpleName() + "s should not be null!"));
+        for (EventAuthorizationDto dto : eventAuthorizationDtos) {
+            String resourceType = dto.getResourceType();
+            Arrays.asList(AuthorizationResourceAction.values()).stream()
+                    .filter(action -> isResourceTypeHasDescribeOrGetAction(action, resourceType))
+                    .findFirst()
+                    .ifPresentOrElse(
+                            action -> checkPermissionForResource(action, dto.getResourceCrn()),
+                            () -> checkHasLegacyActionOrThrow(resourceType)
+                    );
+        }
+    }
+
+    private boolean isResourceTypeHasDescribeOrGetAction(AuthorizationResourceAction action, String resourceType) {
+        return (action.isDescribeAction() || action.isGetAction() || action.isAdminAction())
+                && action.name().contains(resourceType.toUpperCase());
+    }
+
+    private void checkHasLegacyActionOrThrow(String resourceType) {
+        if (!LegacyResourceTypes.getTypeValues().contains(resourceType) && !resourceType.toLowerCase().endsWith("dto")) {
+            throwIllegalStateExceptionForResourceType(resourceType);
+        }
+    }
+
+    private void checkPermissionForResource(AuthorizationResourceAction action, String resourceCrn) {
+        boolean specialActionHasPermission = checkSpecialActionHasPermission(action, resourceCrn);
+        if (!specialActionHasPermission) {
+            checkPermission(action, resourceCrn);
+        }
+    }
+
+    private boolean checkSpecialActionHasPermission(AuthorizationResourceAction action, String resourceCrn) {
```

Comment:
Can you please delete this `actionAlternativeProvider` and solve this with a simple if else?

Reply:
Thank you for the suggestion, but I wouldn't. At this point, it could be more simple that way until we have only one "special action" like the freeipa, but if let's say we have 8 then we should implement 8 different conditions? I think even if there won't be any more special cases in the future like this one, having a separate class that stores/manage these kinds of pairs wouldn't affect nor the readableness nor the performance.

Reply:
As I see this is the only exception therefore I would still use only a simple if-else and not put this hack behind a fancy named component like `AlternativeAuthorizationResourceActionProvider`.